src/scratchOrgSettingsGenerator.ts

```diff
@@ -210,62 +172,168 @@ export default class SettingsGenerator {
-    <types>
-        <members>*</members>
-        <name>Settings</name>
-    </types>
-    <types>
-        <members>*</members>
-        <name>CustomObject</name>
-    </types>
-    <version>${apiVersion}</version>
-</Package>`;
-    this.writer.addToZip(pkg, path.join(this.shapeDirName, 'package.xml'));
-  }
+  private async writePackageFile(
+    allRecordTypes: string[],
+    allbusinessProcesses: string[],
+    packageFilePath: string,
+    apiVersion: string
+  ) {
+    let output = {
+      '@': {
+        xmlns: 'http://soap.sforce.com/2006/04/metadata',
+      },
+      types: [] as JsonArray,
+    };
+    if (this.settingData) {
+      const settingsMemberReferences = Object.keys(this.settingData).reduce((acc, item) => {
+        const typeName = upperFirst(item).replace('Settings', '');
+        return {
+          ...acc,
+          ...{ members: [...(acc.members ?? []), typeName] },
+        };
+      }, Object.create(null));
+      output = { ...output, ...{ types: [...output.types, { ...settingsMemberReferences, name: 'Settings' }] } };
+    }
+    if (this.objectSettingsData) {
+      const objectMemberReferences = Object.keys(this.objectSettingsData).reduce((acc, item) => {
+        return {
+          ...acc,
+          ...{ members: [...(acc.members ?? []), upperFirst(item)] },
+        };
+      }, Object.create(null));
+      output = { ...output, ...{ types: [output.types, { ...objectMemberReferences, name: 'CustomObject' }] } };
+
+      if (allRecordTypes.length > 0) {
+        output = {
+          ...output,
+          ...{
+            types: [
+              output.types,
+              {
+                ...allRecordTypes.reduce((acc, allRecordType) => {
+                  return {
+                    ...acc,
+                    ...{ members: [...(acc.members ?? []), allRecordType] },
+                  };
+                }, Object.create(null)),
+                name: 'RecordType',
+              },
+            ],
+          },
+        };
+      }
 
-  private createObjectFileContent(json: Record<string, unknown>) {
-    const output: ObjectSetting = {};
-    if (json.sharingModel) {
-      output.sharingModel = upperFirst(json.sharingModel as string);
+      if (allbusinessProcesses.length > 0) {
+        output = {
+          ...output,
+          ...{
+            types: [
+              output.types,
+              {
+                ...allbusinessProcesses.reduce((acc, allRecordType) => {
+                  return {
+                    ...acc,
+                    ...{ members: [...(acc.members ?? []), allRecordType] },
+                  };
+                }, Object.create(null)),
+                name: 'BusinessProcess',
+              },
+            ],
+          },
+        };
+      }
     }
-    return output;
+    output = { ...output, ...{ version: apiVersion } };
+    const xml = js2xmlparser.parse('Package', output);
+    this.writer.addToZip(xml, packageFilePath);
```

Comment:
again, imagine how nice the tests for `writePackageFile` could be if it were a pure function that took the inputs it needed, didn't use `this` and returned an object

and let something else convert/write the xml (keeping stuff w/ side effects super simple)